test/common/buffer/owned_impl_test.cc

```diff
@@ -1139,23 +1151,24 @@ TEST_F(OwnedImplTest, PrependEmpty) {
 // Regression test for oss-fuzz issues
 // https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=14466, empty commit
 // following a reserve resulted in a corrupted libevent internal state.
-TEST_F(OwnedImplTest, ReserveZeroCommit) {
+TYPED_TEST(OwnedImplTest, ReserveZeroCommit) {
   BufferFragmentImpl frag("", 0, nullptr);
   Buffer::OwnedImpl buf;
   buf.addBufferFragment(frag);
   buf.prepend("bbbbb");
   buf.add("");
-  expectSlices({{5, 0, 4096}, {0, 0, 0}}, buf);
+  this->expectSlices({{5, 0, 4096}, {0, 0, 0}}, buf);
   { auto reservation = buf.reserveSingleSlice(1280); }
-  expectSlices({{5, 0, 4096}}, buf);
+  this->expectSlices({{5, 0, 4096}}, buf);
   os_fd_t pipe_fds[2] = {0, 0};
   auto& os_sys_calls = Api::OsSysCallsSingleton::get();
 #ifdef WIN32
   ASSERT_EQ(os_sys_calls.socketpair(AF_INET, SOCK_STREAM, 0, pipe_fds).return_value_, 0);
 #else
   ASSERT_EQ(pipe(pipe_fds), 0);
 #endif
-  Network::IoSocketHandleImpl io_handle(pipe_fds[0]);
+  using IoSocketHandleType = typename TestFixture::IoSocketHandleTestType;
```

Comment:
> `Is this the only test where the IO socket impl matters?`

It actually doesn't really matter here both implementations should work on both platforms. That way we have coverage of the Windows code on linux.

> to avoid the typed test stuff everywhere?

This is the first time I am using the typed test pattern. Do they have a lot of gotchas and I am naive in using them? 

> Also potentially just do a compile time switch to only test the windows impl on windows?

In general, I am in favour of testing windows specific implementations on Linux if this is possible because it makes it easier for all developers to debug issues. If this was a Windows-only test then if someone failed it, they would require someone with a Windows machine to run it. While now they can just debug it from their setup.

Reply:
It just makes the test code more verbose. If you only need the IoSocket handle in this test I would just make a TYPED_TEST fixture just for this one test.

Reply:
I separated the tests that rely on IoSocket handle to their own test in e7a10c0399e9549130f762efa0be83b8216c58c1